src/server.ts

```diff
@@ -0,0 +1,348 @@
+		// This is a hack to prevent some weird Travelling Ruby behaviour with Gems
+		// https://github.com/pact-foundation/pact-mock-service-npm/issues/16
+		delete envVars["RUBYGEMS_GEMDEPS"];
+		let file: string;
+		let opts: SpawnOptions = {
+			cwd: pactPath.cwd,
+			detached: !isWindows,
+			env: envVars
+		};
+		let args: string[] = pactUtil.createArguments(this.options, {
+			"port": "--port",
+			"host": "--host",
+			"log": "--log",
+			"ssl": "--ssl",
+			"sslcert": "--sslcert",
+			"sslkey": "--sslkey",
+			"cors": "--cors",
+			"dir": "--pact_dir",
+			"spec": "--pact_specification_version",
+			"consumer": "--consumer",
+			"provider": "--provider"
+		});
+
+		let cmd: string = [pactPath.file].concat(args).join(" ");
+
+		if (isWindows) {
+			file = "cmd.exe";
+			args = ["/s", "/c", cmd];
+			(opts as any).windowsVerbatimArguments = true;
+		} else {
+			cmd = `./${cmd}`;
+			file = "/bin/sh";
+			args = ["-c", cmd];
+		}
+		logger.debug(`Starting binary with '${_.flatten([file, args, JSON.stringify(opts)])}'`);
+		this.__instance = cp.spawn(file, args, opts);
+
+		this.__instance.stdout.setEncoding("utf8");
+		this.__instance.stdout.on("data", logger.debug.bind(logger));
+		this.__instance.stderr.setEncoding("utf8");
+		this.__instance.stderr.on("data", logger.debug.bind(logger));
+		this.__instance.on("error", logger.error.bind(logger));
+
+		// if port isn't specified, listen for it when pact runs
+		const catchPort = (data) => {
+			const match = data.match(/port=([0-9]+)/);
+			if (match && match[1]) {
+				this.options.port = parseInt(match[1], 10);
+				this.__instance.stdout.removeListener("data", catchPort.bind(this));
+				this.__instance.stderr.removeListener("data", catchPort.bind(this));
+				logger.info(`Pact running on port ${this.options.port}`);
+			}
+		};
+
+		if (!this.options.port) {
+			this.__instance.stdout.on("data", catchPort.bind(this));
+			this.__instance.stderr.on("data", catchPort.bind(this));
+		}
+
+		logger.info(`Creating Pact with PID: ${this.__instance.pid}`);
+
+		this.__instance.once("close", (code) => {
+			if (code !== 0) {
+				logger.warn(`Pact exited with code ${code}.`);
+			}
+			return this.stop();
+		});
+
+		// check service is available
+		return this.__waitForServerUp(this.options)
+			.timeout(PROCESS_TIMEOUT, `Couldn't start Pact with PID: ${this.__instance.pid}`)
+			.then(() => {
+				this.__running = true;
+				this.emit(Server.Events.START_EVENT, this);
+				return this;
+			});
+	}
+
+	// Stop the server instance, no more mocking
+	public stop(): q.Promise<Server> {
+		let pid = -1;
+		if (this.__instance) {
+			pid = this.__instance.pid;
+			logger.info(`Removing Pact with PID: ${pid}`);
+			this.__instance.removeAllListeners();
+			// Killing instance, since windows can't send signals, must kill process forcefully
+			try {
+				if (isWindows) {
+					cp.execSync(`taskkill /f /t /pid ${pid}`);
+				} else {
+					process.kill(-pid, "SIGINT");
```

Comment:
What's the reason for the `-pid` rather than just `pid`?

Reply:
it's how unix system does sigint, but specifying the negative PID.  Not sure why, but it works so...